### ISSUE TITLE
Compilation error on ppc64le for branch next

### ISSUE THREAD

**Issue by ggeorgakoudis** [boutproject/BOUT-dev#2986]
Compilation on a ppc64le machine (GCC 12.2.1, CUDA 12.2.2) fails with this error message:
```bash
/usr/workspace/ggeorgak/projects/boutdev/BOUT-dev/include/bout/parallel_boundary_region.hxx(99): error: no instance of overloaded function "std::vector<_Tp, _Alloc>::push_back [with _Tp=BoundaryRegionPar::Indices, _Alloc=std::allocator<BoundaryRegionPar::Indices>]" matches the argument list
            argument types are: ({...})
            object type is: BoundaryRegionPar::IndicesVec
      bndry_points.push_back({ind, {x, y, z}, length, valid});
                   ^

/usr/workspace/ggeorgak/projects/boutdev/BOUT-dev/include/bout/parallel_boundary_region.hxx(103): error: no instance of overloaded function "std::vector<_Tp, _Alloc>::push_back [with _Tp=BoundaryRegionPar::Indices, _Alloc=std::allocator<BoundaryRegionPar::Indices>]" matches the argument list
            argument types are: ({...})
            object type is: BoundaryRegionPar::IndicesVec
      bndry_points.push_back({xyz2ind(ix, iy, iz, localmesh), {x, y, z}, length, valid});
                   ^

2 errors detected in the compilation of "/usr/workspace/ggeorgak/projects/boutdev/BOUT-dev/src/field/field3d.cxx".
```

The problem is that we call the implicit constructor of https://github.com/boutproject/BOUT-dev/blob/next/include/bout/parallel_boundary_region.hxx#L63-L74, expecting `signed char` for the field `valid`.

However, callers, e.g., https://github.com/boutproject/BOUT-dev/blob/next/include/bout/parallel_boundary_region.hxx#L99, use a `char` value in the initializer list.

This is error-free on x86 because the types `signed char` and `char` are equivalent (see https://en.cppreference.com/w/cpp/language/types), but it is a narrowing error on ppc64le where `char` equates `unsigned char`.

What is the original intention on using `signed char`, are negative values meaningful? Depending on the answer, we will need to fix this issue by changing the type.

Ping @dschwoerer as the merger of the related commit.

**Comment by bendudson** [comment]
Thanks @ggeorgakoudis ! The [struct definition](https://github.com/boutproject/BOUT-dev/blob/next/include/bout/parallel_boundary_region.hxx#L63-L74) suggests that `valid` is a count:
```
struct Indices {
  ...
  // How many points we can go in the opposite direction
  signed char valid;
};
```
so I would think it should be `>=0` unless negative numbers are being used as error conditions. I suspect this should be `unsigned char`in this definition and the interface functions. Even better might be `size_t`. I don't know the intent here, but perhaps @dschwoerer does.

**Comment by dschwoerer** [comment]
Previous build failed to start:
https://copr.fedorainfracloud.org/coprs/davidsch/testing-all/build/8117131/

**Comment by ggeorgakoudis** [comment]
Closed by PR #2987
